src/trajectory.jl

```diff
@@ -202,6 +218,20 @@ end
 ### The doubling tree algorithm for expanding trajectory.
 ###
 
+"""
+Divergence reasons
+- `dynamic`: due to stoping criteria
+- `numerical`: due to large energy deviation from starting (possibly numerical errors)
+"""
+struct Divergence
```

Comment:
Maybe rename `Divergence` to `Termination`, since `u-turn` is not a divergence? Also, we can use `Enums` here, e.g.

```julia

@enum Termination default=0 numerical=1 dynamic=2

isdivergent(d::Termination) = Int(d) > 0
```

This allows us to add other `termination` types easily in the future -- no need to add new fields to the `Termination` type.

See https://docs.julialang.org/en/v1/base/base/#Base.Enums.@enum

Reply:
This makes sense. I was trying to allow scenario where both numerical and dynamic could be true but there seems no such case.